mod-clans/src/main/java/gg/solarmc/loader/clans/Clan.java

```diff
@@ -0,0 +1,407 @@
+        if (this.getAlliedClan(transaction).isPresent()) throw new IllegalStateException("Sender already have an ally!");
+        if (receiver.getAlliedClan(transaction).isPresent()) throw new IllegalStateException("Selected clan already has an ally!");
+
+        int res = transaction.getProperty(DSLContext.class)
+                .insertInto(CLANS_CLAN_ALLIANCES)
+                .columns(CLANS_CLAN_ALLIANCES.CLAN_ID,CLANS_CLAN_ALLIANCES.ALLY_ID)
+                .values(this.clanID,receiver.getID())
+                .execute();
+
+        this.alliedClan = receiver.getID();
+        receiver.alliedClan = this.getID();
+
+        return res == 1;
+    }
+
+    /**
+     * Revokes ally. Can be called from either clan. Revokes alliance for both.
+     * @param transaction the tx
+     * @return whether operation was successful or not
+     */
+    public boolean revokeAlly(Transaction transaction) {
+
+        int i = transaction.getProperty(DSLContext.class)
+                .deleteFrom(CLANS_CLAN_ALLIANCES)
+                .where(CLANS_CLAN_ALLIANCES.CLAN_ID.eq(this.clanID))
+                .or(CLANS_CLAN_ALLIANCES.ALLY_ID.eq(this.clanID))
+                .execute();
+
+        this.getAlliedClan(transaction).ifPresent(c -> c.alliedClan = null);
+        this.alliedClan = null;
+
+        return i == 1;
+    }
+
+    /**
+     * Marks clan as an enemy
+     * @param transaction twix bar
+     * @param receiver the clan to mark as an enemy.
+     * @return whether action was successful or not (failure represents that they are already enemies)
+     * @throws IllegalArgumentException if the presented clan is an ally or is the same object (what?)
+     */
+    public boolean addClanAsEnemy(Transaction transaction, Clan receiver) {
+        if (receiver.getID() == this.clanID) throw new IllegalArgumentException("Tried to mark clan enemy as same clan!");
+
+        this.getAlliedClan(transaction).ifPresent(ally -> {
+            if (this.alliedClan.equals(ally.getID())) throw new IllegalArgumentException("Tried to add allied clan as enemy!");
+        });
+
+        int res = transaction.getProperty(DSLContext.class)
+                .insertInto(CLANS_CLAN_ENEMIES)
+                .columns(CLANS_CLAN_ENEMIES.CLAN_ID,CLANS_CLAN_ENEMIES.ENEMY_ID)
+                .values(this.clanID,receiver.getID())
+                .execute();
+
+        return res == 1;
+    }
+
+    /**
+     * Marks clan enemy as no longer an enemy
+     * @param transaction the tx
+     * @param clan clan provided
+     * @throws IllegalArgumentException if passed self or if passed ally
+     */
+    public void removeClanAsEnemy(Transaction transaction, Clan clan) {
+        if (clan.getID() == this.clanID) throw new IllegalArgumentException("Tried to unmark self as enemy??");
+
+        this.getAlliedClan(transaction).ifPresent(ally -> {
+            if (this.alliedClan.equals(ally.getID())) throw new IllegalArgumentException("Tried to unmark allied clan as enemy!");
```

Comment:
```suggestion
            if (clan.getID() == ally.getID()) throw new IllegalArgumentException("Tried to unmark allied clan as enemy!");
```
I'm pretty sure this is what you meant to do

Reply:
Fixed!